Update description for error E0308 (#78332)
rust-lang/rust: compiler/rustc_error_codes/src/error_codes/E0308.md

> Erroneous code example:

Reviewer: Since there are multiple examples:

```suggestion
Erroneous code examples:
```

```compile_fail,E0308
fn plus_one(x: i32) -> i32 {
    x + 1
}

plus_one("Not a number");
//       ^^^^^^^^^^^^^^ expected `i32`, found `&str`

if "Not a bool" {
// ^^^^^^^^^^^^ expected `bool`, found `&str`

> }

Reviewer: Could you also add something similar to the original example? That way there's an example that explicitly shows a type mismatch as well.

```

This error occurs when an expression was used in a place where the compiler
expected an expression of a different type. It can occur in several cases, the

> most common being when calling a function, and passing an argument which has a

Reviewer: The comma makes it look like you're referring to two separate cases.

```suggestion
most common being when calling a function and passing an argument which has a
```

> different type than the matching argument in the function declaration.

Reviewer: Something like this maybe?

```suggestion
different type than the type declared in the function declaration.
```

Author: "type declared in the function declaration" sounds awkward to me

Reviewer: The way it is currently sounds a little funny to me, but it's not a big deal.
